src/bin/main.rs

```diff
@@ -375,6 +385,27 @@ fn sender_print_code(term: &mut Term, code: &magic_wormhole::Code) -> eyre::Resu
     Ok(())
 }
 
+fn execute_shell_command(
+    term: &mut Term,
+    matches: &clap::ArgMatches<'_>,
+    code: &magic_wormhole::Code
+) -> eyre::Result<()> {
+    if let Some(shell_command) = matches.value_of_os("shell-command") {
+        let mut child = std::process::Command::new("/bin/sh")
+            .arg("-c")
+            .arg(shell_command)
+            .stdin(std::process::Stdio::piped())
+            .spawn()
+            .expect("failed execute shell-command");
```

Comment:
Error handling instead of panic:

```suggestion
            .context("failed execute shell-command")?;
```

(same applies for the lines below too)

Reply:
Had to include some things from the `eyre` namespace.